In [1]:
import pandas as pd


In [2]:
results = pd.read_csv("../data/raw/results.csv")

In [3]:
results.head()

,date,home_team,away_team,home_score,away_score,tournament,city,country,neutral
0,1872-11-30,Scotland,England,0.0,0.0,Friendly,Glasgow,Scotland,False
1,1873-03-08,England,Scotland,4.0,2.0,Friendly,London,England,False
2,1874-03-07,Scotland,England,2.0,1.0,Friendly,Glasgow,Scotland,False
3,1875-03-06,England,Scotland,2.0,2.0,Friendly,London,England,False
4,1876-03-04,Scotland,England,3.0,0.0,Friendly,Glasgow,Scotland,False


In [4]:
results.shape

(49499, 9)

In [5]:
results.info()

<class 'pandas.DataFrame'>
RangeIndex: 49499 entries, 0 to 49498
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   date        49499 non-null  str    
 1   home_team   49499 non-null  str    
 2   away_team   49499 non-null  str    
 3   home_score  49490 non-null  float64
 4   away_score  49490 non-null  float64
 5   tournament  49499 non-null  str    
 6   city        49499 non-null  str    
 7   country     49499 non-null  str    
 8   neutral     49499 non-null  bool   
dtypes: bool(1), float64(2), str(6)
memory usage: 5.9 MB


In [6]:
results.describe(include="all")

,date,home_team,away_team,home_score,away_score,tournament,city,country,neutral
count,49499,49499,49499,49490.000000,49490.000000,49499,49499,49499,49499
unique,16480,327,321,NaN,NaN,200,2089,269,2
top,2012-02-29,Brazil,Uruguay,NaN,NaN,Friendly,Kuala Lumpur,United States,False
freq,66,618,585,NaN,NaN,18388,745,1576,36360
mean,NaN,NaN,NaN,1.757345,1.181815,NaN,NaN,NaN,NaN
std,NaN,NaN,NaN,1.773916,1.401670,NaN,NaN,NaN,NaN
min,NaN,NaN,NaN,0.000000,0.000000,NaN,NaN,NaN,NaN
25%,NaN,NaN,NaN,1.000000,0.000000,NaN,NaN,NaN,NaN
50%,NaN,NaN,NaN,1.000000,1.000000,NaN,NaN,NaN,NaN
75%,NaN,NaN,NaN,2.000000,2.000000,NaN,NaN,NaN,NaN


## Initial Data Exploration

Key findings:

- Dataset contains 49,499 international matches.
- Data spans over 150 years.
- Only 9 matches have missing scores.
- Home teams score an average of 1.76 goals compared to 1.18 for away teams.
- Friendly matches account for the largest share of the dataset.
- 327 unique national teams are represented.

In [7]:
former_names = pd.read_csv("../data/raw/former_names.csv")

In [8]:
former_names.head()

,current,former,start_date,end_date
0,Benin,Dahomey,1959-11-08,1975-11-30
1,Burkina Faso,Upper Volta,1960-04-14,1984-08-04
2,Curaçao,Netherlands Antilles,1957-03-03,2010-10-10
3,Czechoslovakia,Bohemia,1903-04-05,1919-01-01
4,Czechoslovakia,Bohemia and Moravia,1939-01-01,1945-05-01


In [9]:
former_names.info()

<class 'pandas.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype
---  ------      --------------  -----
 0   current     36 non-null     str  
 1   former      36 non-null     str  
 2   start_date  36 non-null     str  
 3   end_date    36 non-null     str  
dtypes: str(4)
memory usage: 2.7 KB


In [10]:
former_names.shape

(36, 4)

In [11]:
shootouts = pd.read_csv("../data/raw/shootouts.csv")

In [12]:
shootouts.head()

,date,home_team,away_team,winner,first_shooter
0,1967-08-22,India,Taiwan,Taiwan,NaN
1,1971-11-14,South Korea,Vietnam Republic,South Korea,NaN
2,1972-05-07,South Korea,Iraq,Iraq,NaN
3,1972-05-17,Thailand,South Korea,South Korea,NaN
4,1972-05-19,Thailand,Cambodia,Thailand,NaN


In [13]:
shootouts.info()
shootouts.shape
shootouts.describe(include="all")

<class 'pandas.DataFrame'>
RangeIndex: 680 entries, 0 to 679
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype
---  ------         --------------  -----
 0   date           680 non-null    str  
 1   home_team      680 non-null    str  
 2   away_team      680 non-null    str  
 3   winner         680 non-null    str  
 4   first_shooter  258 non-null    str  
dtypes: str(5)
memory usage: 51.6 KB


,date,home_team,away_team,winner,first_shooter
count,680,680,680,680,258
unique,594,188,198,183,96
top,2016-06-03,South Africa,Egypt,South Korea,Colombia
freq,5,18,16,15,11


In [14]:
goalscorers = pd.read_csv("../data/raw/goalscorers.csv")

In [15]:
goalscorers.head()

,date,home_team,away_team,team,scorer,minute,own_goal,penalty
0,1916-07-02,Chile,Uruguay,Uruguay,José Piendibene,44.0,False,False
1,1916-07-02,Chile,Uruguay,Uruguay,Isabelino Gradín,55.0,False,False
2,1916-07-02,Chile,Uruguay,Uruguay,Isabelino Gradín,70.0,False,False
3,1916-07-02,Chile,Uruguay,Uruguay,José Piendibene,75.0,False,False
4,1916-07-06,Argentina,Chile,Argentina,Alberto Ohaco,2.0,False,False


In [16]:
goalscorers.shape
goalscorers.describe(include="all")


,date,home_team,away_team,team,scorer,minute,own_goal,penalty
count,47855,47855,47855,47855,47811,47601.000000,47855,47855
unique,4749,220,220,220,15384,NaN,2,2
top,2011-10-11,Brazil,Uruguay,Brazil,Cristiano Ronaldo,NaN,False,False
freq,145,1054,803,1084,124,NaN,46921,44595
mean,NaN,NaN,NaN,NaN,NaN,50.059999,NaN,NaN
std,NaN,NaN,NaN,NaN,NaN,26.369945,NaN,NaN
min,NaN,NaN,NaN,NaN,NaN,1.000000,NaN,NaN
25%,NaN,NaN,NaN,NaN,NaN,28.000000,NaN,NaN
50%,NaN,NaN,NaN,NaN,NaN,51.000000,NaN,NaN
75%,NaN,NaN,NaN,NaN,NaN,73.000000,NaN,NaN


In [17]:
goalscorers.info()

<class 'pandas.DataFrame'>
RangeIndex: 47855 entries, 0 to 47854
Data columns (total 8 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   date       47855 non-null  str    
 1   home_team  47855 non-null  str    
 2   away_team  47855 non-null  str    
 3   team       47855 non-null  str    
 4   scorer     47811 non-null  str    
 5   minute     47601 non-null  float64
 6   own_goal   47855 non-null  bool   
 7   penalty    47855 non-null  bool   
dtypes: bool(2), float64(1), str(5)
memory usage: 4.5 MB
